src/pages/ThemeProductPage/ThemeProductList.tsx

```diff
@@ -99,16 +44,24 @@ const ThemeProductList = () => {
     observer.observe(observerTarget);
 
     return () => observer.disconnect();
-  }, [loadProducts, hasMore, loading]);
+  }, [fetchNextPage, hasNextPage, isLoading]);
 
-  if (!products.length && !loading) {
+  if (isError) {
+    console.error('상품 로딩 실패:', error);
+    navigate('/');
+    return null;
```

Comment:
[제안] react router가 제공하는 `Navigate` 컴포넌트로 얼리 리턴하는 방식을 사용하면 더 간결할 것 같아요

```suggestion
    return <Navigate ... />
```

Reply:
navigate('/') 대신     return <Navigate to={''} />  방식으로 리팩터링했습니다.